service/src/main/java/bio/terra/cbas/dependencies/wes/CromwellClient.java

```diff
@@ -1,34 +1,69 @@
 package bio.terra.cbas.dependencies.wes;
 
+import bio.terra.cbas.common.exceptions.AzureAccessTokenException;
+import bio.terra.cbas.common.exceptions.DependencyNotAvailableException;
 import bio.terra.cbas.config.CromwellServerConfiguration;
+import bio.terra.cbas.dependencies.common.CredentialLoader;
+import bio.terra.cbas.dependencies.common.DependencyUrlLoader;
 import cromwell.client.ApiClient;
 import cromwell.client.api.EngineApi;
 import cromwell.client.api.Ga4GhWorkflowExecutionServiceWesAlphaPreviewApi;
 import java.util.Optional;
+import okhttp3.OkHttpClient;
 import org.springframework.stereotype.Component;
 
 @Component
 public class CromwellClient {
 
   private final CromwellServerConfiguration cromwellServerConfiguration;
+  private final DependencyUrlLoader dependencyUrlLoader;
+  private final CredentialLoader credentialLoader;
 
-  public CromwellClient(CromwellServerConfiguration cromwellServerConfiguration) {
+  private final OkHttpClient singletonHttpClient;
+
+  public CromwellClient(
+      CromwellServerConfiguration cromwellServerConfiguration,
+      DependencyUrlLoader dependencyUrlLoader,
+      CredentialLoader credentialLoader) {
     this.cromwellServerConfiguration = cromwellServerConfiguration;
+    this.dependencyUrlLoader = dependencyUrlLoader;
+    this.credentialLoader = credentialLoader;
+    singletonHttpClient = new ApiClient().getHttpClient();
   }
 
-  private ApiClient getApiClient() {
-    return new ApiClient().setBasePath(cromwellServerConfiguration.baseUri());
+  private ApiClient getApiClient()
+      throws DependencyNotAvailableException, AzureAccessTokenException {
+    String uri;
+
+    String baseUriFromConfig = cromwellServerConfiguration.baseUri();
+    if (!baseUriFromConfig.isEmpty()) {
+      uri = baseUriFromConfig;
+    } else {
+      uri = dependencyUrlLoader.loadDependencyUrl(DependencyUrlLoader.DependencyUrlType.CROMWELL);
+    }
```

Comment:
So this is going to be a little different from the other lookups.

For _read_ operations we will always want to use the URL in config, for _write_ operations we will always want to look up the right per-user cromwell instance.